File: C/7 kyu/7 kyu - Easy wallpaper.c
// https://www.codewars.com/kata/easy-wallpaper/train/r
// My solution
#include <math.h>

const char* wallPaper(double l, double w, double h)
{
    static const char *const numbers[21] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten",
    "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen", "twenty"};
    
    if(l == 0 || w == 0 || h == 0) return numbers[0];

    double extra_length = 1.15; // percent
    l *= extra_length;
    w *= extra_length;
    double wall_area = ( l * h * 2 ) + ( w * h * 2 ); // square meters

    double standard_width_of_roll = 0.52;  // meters
    double standard_length_of_roll = 10.0; // meters
    double roll_area = standard_width_of_roll * standard_length_of_roll; // square meters
    
    int rolls = ceil(wall_area / roll_area);
    
    return numbers[rolls];
}

// ...
#include <math.h>

char* wallPaper(double l, double w, double h)
{
    char *numbers[21] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten",
    "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen", "twenty"};
    if (l == 0 || w == 0 || h == 0)
        return "zero";
    double r = (( (l + w) * 2 * h) / 5.2) * 1.15;
    return numbers[(long)ceil(r)];
}

File: R/7 kyu - Easy wallpaper.R
# https://www.codewars.com/kata/easy-wallpaper/train/r
# My solution
numbers_0_20 = c("zero","one","two","three","four","five",
                  "six","seven","eight","nine","ten",
                  "eleven","twelve","thirteen","fourteen","fifteen",
                  "sixteen","seventeen","eighteen","nineteen","twenty")

more_15_percent <- function (value) {
    return(value + value * 0.15)
}

wallpaper <- function (length, width, height) {
    wall_in_square_meters = height * length * 2 + height * width * 2
    wall_in_square_meters = more_15_percent(wall_in_square_meters)
    rolls = wall_in_square_meters/5.2 + 1
    return(numbers_0_20[ceiling(rolls)])
}

# ...
wallpaper <- function (l, w, h) {
    switch(ceiling(2.3*h*(l + w) / 5.2) + 1,
        "zero", "one", "two", "three", "four",
        "five", "six", "seven", "eight", "nine",
        "ten", "eleven", "twelve", "thirteen", "fourteen",
        "fifteen", "sixteen", "seventeen", "eighteen", "nineteen",
        "twenty")
}

# ...
wallpaper <- function (l, w, h) {
  numbers<-c("one","two","three","four","five","six","seven","eight","nine","ten","eleven","twelve","thirteen","fourteen","fifteen","sixteen","seventeen","eighteen","nineteen","twenty")
  numbers[trunc(2*(l+w)*h/.52*1.15/10)+1]
}